scripts/alertmailer.php

```diff
@@ -30,8 +30,32 @@ function mlog($message) {
 if (!$lastbatch) $lastbatch = 0;
 mlog("lastupdated: $lastupdated lastbatch: $lastbatch\n");
 
-# Construct query fragment to select search index batches which
-# have been made since last time we ran
+
+// extract html wrappers from html email template
+// tokens have been added to the template to mark the start and end of section of the html
+$html_template_filename=HTML_EMAIL_TEMPLATE; 
+if (!file_exists($html_template_filename)) {
+    $PAGE->error_message("Sorry, we could not find the email template '" . $filename . "'.");
+    return false; }
+mlog(HTML_EMAIL_TEMPLATE);
+// Get the text from the template.
+$content = file_get_contents($html_template_filename);
+//content, start_token, end_token
+$html_email_sections['TOP']=extract_content_between_tokens($content,'<!-- CUT_HERE START_TOP -->','<!-- CUT_HERE END_TOP -->',false);
+$html_email_sections['MEMBER_HEADER']=extract_content_between_tokens($content,'<!-- CUT_HERE START_MEMBER_SEARCH_HEADER -->','<!-- CUT_HERE END_MEMBER_SEARCH_HEADER -->',true);
+$html_email_sections['MEMBER_ITEM']=extract_content_between_tokens($content,'<!-- CUT_HERE START_MEMBER_WRAPPER -->','<!-- CUT_HERE END_MEMBER_WRAPPER -->',true);
+$html_email_sections['UNSUB_ALERT']=extract_content_between_tokens($content,'<!-- CUT_HERE START_END_OF_SECTION_UNSUBSCRIBE -->','<!-- CUT_HERE END_END_OF_SECTION_UNSUBSCRIBE -->',true);
+$html_email_sections['VIEW_MORE']=extract_content_between_tokens($content,'<!-- CUT_HERE START_VIEW_MORE_LINK -->','<!-- CUT_HERE END_VIEW_MORE_LINK -->',true);
+$html_email_sections['PHRASE_HEADER']=extract_content_between_tokens($content,'<!-- CUT_HERE START_PHRASE_SEARCH_HEADER -->','<!-- CUT_HERE END_PHRASE_SEARCH_HEADER -->',true);
+$html_email_sections['PHRASE_ITEM']=extract_content_between_tokens($content,'<!-- CUT_HERE START_PHRASE_WRAPPER -->','<!-- CUT_HERE END_PHRASE_WRAPPER -->',true);
+$html_email_sections['ALERT_PREFS']=extract_content_between_tokens($content,'<!-- CUT_HERE START_NEWSLETTER_PREFERENCES_MANAGEMENT -->','<!-- CUT_HERE END_NEWSLETTER_PREFERENCES_MANAGEMENT -->',true);
+$html_email_sections['SEPERATION_LINE']=extract_content_between_tokens($content,'<!-- CUT_HERE START_SEPERATION_LINE -->','<!-- CUT_HERE END_SEPERATION_LINE -->',true);
+$html_email_sections['BOTTOM']=extract_content_between_tokens($content,'<!-- CUT_HERE START_BOTTOM -->','<!-- CUT_HERE END_BOTTOM -->',false);
+//mlog($html_email_sections['MEMBER_HEADER']);
```

Comment:
There will be a few of these, so perhaps a search and remove of lines
starting with '//mlog'

On Sun, Feb 15, 2015 at 3:06 PM, Henare Degan <notifications@github.com>
wrote:

> In scripts/alertmailer.php
> <https://github.com/openaustralia/twfy/pull/11#discussion_r24719526>:
>
> > +    return false; }
> > +mlog(HTML_EMAIL_TEMPLATE);
> > +// Get the text from the template.
> > +$content = file_get_contents($html_template_filename);
> > +//content, start_token, end_token
> > +$html_email_sections['TOP']=extract_content_between_tokens($content,'<!-- CUT_HERE START_TOP -->','<!-- CUT_HERE END_TOP -->',false);
> > +$html_email_sections['MEMBER_HEADER']=extract_content_between_tokens($content,'<!-- CUT_HERE START_MEMBER_SEARCH_HEADER -->','<!-- CUT_HERE END_MEMBER_SEARCH_HEADER -->',true);
> > +$html_email_sections['MEMBER_ITEM']=extract_content_between_tokens($content,'<!-- CUT_HERE START_MEMBER_WRAPPER -->','<!-- CUT_HERE END_MEMBER_WRAPPER -->',true);
> > +$html_email_sections['UNSUB_ALERT']=extract_content_between_tokens($content,'<!-- CUT_HERE START_END_OF_SECTION_UNSUBSCRIBE -->','<!-- CUT_HERE END_END_OF_SECTION_UNSUBSCRIBE -->',true);
> > +$html_email_sections['VIEW_MORE']=extract_content_between_tokens($content,'<!-- CUT_HERE START_VIEW_MORE_LINK -->','<!-- CUT_HERE END_VIEW_MORE_LINK -->',true);
> > +$html_email_sections['PHRASE_HEADER']=extract_content_between_tokens($content,'<!-- CUT_HERE START_PHRASE_SEARCH_HEADER -->','<!-- CUT_HERE END_PHRASE_SEARCH_HEADER -->',true);
> > +$html_email_sections['PHRASE_ITEM']=extract_content_between_tokens($content,'<!-- CUT_HERE START_PHRASE_WRAPPER -->','<!-- CUT_HERE END_PHRASE_WRAPPER -->',true);
> > +$html_email_sections['ALERT_PREFS']=extract_content_between_tokens($content,'<!-- CUT_HERE START_NEWSLETTER_PREFERENCES_MANAGEMENT -->','<!-- CUT_HERE END_NEWSLETTER_PREFERENCES_MANAGEMENT -->',true);
> > +$html_email_sections['SEPERATION_LINE']=extract_content_between_tokens($content,'<!-- CUT_HERE START_SEPERATION_LINE -->','<!-- CUT_HERE END_SEPERATION_LINE -->',true);
> > +$html_email_sections['BOTTOM']=extract_content_between_tokens($content,'<!-- CUT_HERE START_BOTTOM -->','<!-- CUT_HERE END_BOTTOM -->',false);
> > +//mlog($html_email_sections['MEMBER_HEADER']);
>
> Remove debugging line?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openaustralia/twfy/pull/11/files#r24719526>.
>



-- 
+61 468 335597
*LinkedIn Profile : Matt Joyce
<http://au.linkedin.com/pub/matthew-joyce/6/4b5/6a4/>*